tests/unit/shutdown.tcl

```diff
@@ -3,12 +3,16 @@ start_server {tags {"shutdown external:skip"}} {
         for {set i 0} {$i < 20} {incr i} {
             r set $i $i
         }
-        # It will cost 2s(20 * 100ms) to dump rdb
-        r config set rdb-key-save-delay 100000
+        r config set rdb-key-save-delay 10000000
 
         # Child is dumping rdb
         r bgsave
-        after 100
```

Comment:
so maybe we should have both the wait_for_condition and the `after 100`, in order to attempt to cover case 1 in your description?

Reply:
i meant after the wait_for_condition, so that it'll wait for the file to get created.
i made the change and added a comment

Reply:
ohh, right. a little sleepy...